mypy/checkmember.py

```diff
@@ -778,6 +778,13 @@ def analyze_class_attribute_access(itype: Instance,
         if not mx.is_lvalue:
             result = analyze_descriptor_access(mx.original_type, result, mx.builtin_type,
                                                mx.msg, mx.context, chk=mx.chk)
+
+        # Call the class attribute hook before returning.
+        fullname = '{}.{}'.format(info.fullname, name)
+        hook = mx.chk.plugin.get_class_attribute_hook(fullname)
+        if hook:
+            result = hook(AttributeContext(get_proper_type(mx.original_type),
+                                           result, mx.context, mx.chk))
```

Comment:
I wasn't sure if the hook should be invoked for every exit point of this function. If so, I can pull the hook invocation into a wrapper function that calls into this one (or similar). I'm not familiar enough with this code to be confident where it would and would not make sense to use the hook, so advice is appreciated!

Reply:
Just so I understand, calling through to another function like this is fine, but doing the "opposite" is unoptimized? Though, I realize I wasn't super clear in my previous comment haha. To be a bit more explicit, what I was suggesting was to rename `analyze_class_attribute_access` to something like `analyze_class_attribute_access_inner` and then create a new
```python
def analyze_class_attribute_access(...):
    result = analyze_class_attribute_access_inner(...)
    fullname = '{}.{}'.format(info.fullname, name)
    hook = mx.chk.plugin.get_class_attribute_hook(fullname)
    if hook:
        result = hook(AttributeContext(get_proper_type(mx.original_type),
                                       result, mx.context, mx.chk))
```
Would that pattern be unoptimized, or is it functions/closures declared *within* a function that's unoptimized?

In any case, when looking through this a bit more it does seem there's maybe a few exit points we *don't* want to call the hook for (which I wanted to double check with you) in addition to the `None` return, namely these two cases, which might make the whole `_inner` idea above moot anyway:

https://github.com/python/mypy/blob/6e8c0cd3ef7a375e143faf237312004adcfca7d6/mypy/checkmember.py#L782-L789

Do we want to call the hook for these two exit points?